core/jobs/batch_jobs/question_migration_jobs.py

```diff
@@ -0,0 +1,270 @@
+            object or Exception. Question summary object is returned when the
+            migration was successful and Exception is returned otherwise.
+        """
+        try:
+            question_summary = (
+                question_services.get_question_summary_from_model(
+                question_summary_model
+                )
+            )
+            question_summary.validate()
+        except Exception as e:
+            logging.exception(e)
+            return result.Err((question_summary_id, e))
+
+        return result.Ok((question_summary_id, question_summary))
+
+    @staticmethod
+    def _update_question_summary(
+        question: question_domain.Question,
+        question_summary: question_domain.QuestionSummary,
+        question_summary_model: question_models.QuestionSummaryModel
+    ) -> question_models.QuestionSummaryModel:
+        """Adds version to get an updated question summary model.
+
+        Args:
+            question: Question. The question domain object.
+            question_summary: QuestionSummary. The question summary domain
+                object.
+            question_summary_model: QuestionSummaryModel. The question summary
+                model to update.
+
+        Returns:
+            QuestionSummaryModel. The updated question summary model to put
+            into the datastore.
+        """
+        question_summary_dict = {
+            'interaction_id': question_summary.interaction_id,
+            'question_content': question_summary.question_content,
+            'question_model_last_updated': question_summary.last_updated,
+            'question_model_created_on': question_summary.created_on,
+            'misconception_ids': question_summary.misconception_ids,
+            'version': question.version
+        }
+
+        if question_summary_model is not None:
+            question_summary_model.populate(**question_summary_dict)
+        else:
+            question_summary_dict['id'] = question_summary.id
+            question_summary_model = question_models.QuestionSummaryModel(
+                **question_summary_dict)
+
+        return question_summary_model
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of results from the question migration.
+
+        Returns:
+            PCollection. A PCollection of results from the question
+            migration.
+        """
+        all_question_models = (
+            self.pipeline
+            | 'Get all non-deleted question models' >> (
+                ndb_io.GetModels(question_models.QuestionModel.get_all()))
+            # Pylint disable is needed becasue pylint is not able to correclty
+            # detect that the value is passed through the pipe.
+            | 'Add question keys' >> beam.WithKeys( # pylint: disable=no-value-for-parameter
+                lambda question_model: question_model.id)
+        )
```

Comment:
Should we just have the `{ model_id: version }` from the question models here?

Reply:
Sorry for the misunderstanding! I'm was wondering why we need to carry the entire model from this pipeline if we only have to use the version number later. SO instead can we do something around:

```
question_id_to_version = (
            self.pipeline
            | 'Get all non-deleted question models' >> (
                ndb_io.GetModels(question_models.QuestionModel.get_all()))
            # Pylint disable is needed becasue pylint is not able to correclty
            # detect that the value is passed through the pipe.
            | 'Add question keys' >> beam.WithKeys( # pylint: disable=no-value-for-parameter
                lambda question_model: question_model.id)
            | 'Extract version number' >> beam. Map( lambda m_id, model: m_id, model. Version)
        )
```

Reply:
Done